libpromises/evalfunction.c

```diff
@@ -1188,6 +1188,33 @@ static FnCallResult FnCallGetGid(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const P
 #endif
 }
 
+/*********************************************************************/ 
+
+static FnCallResult FnCallUserInGroup(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, ARG_UNUSED const FnCall *fp, const Rlist *finalargs)
+{
+    assert(finalargs != NULL);
+
+    const char *user_name = RlistScalarValue(finalargs);
+    const char *group_name = RlistScalarValue(finalargs->next);
+
+    StringSet *groups = StringSetNew();
+    
+    struct group *grent = getgrnam(group_name);
```

Comment:
It would be nice if you checked for error. How do you differentiate between error and no entry? You can set
```C
errno = 0;
```
before the call to `getgrnam()`. And then check the value of `errno` after the call. Check man page to see which error codes means that the given name or gid was not found. For all other codes, you can print an error message.